File: CodeSignal/array_replace.rb
# https://app.codesignal.com/arcade/intro/level-6/mCkmbxdMsMTjBc3Bm

def arrayReplace(array, elemToReplace, substitutionElem)
  for el in (0...array.length) 
      array[el] = substitutionElem if array[el] == elemToReplace 
  end 
  array 
end


# second  

def arrayReplace(array, elemToReplace, substitutionElem)
  array.map { |x| x == elemToReplace ? substitutionElem : x } 
end

File: Leetcode/Java/ArrayReplace.java
// https://app.codesignal.com/arcade/intro/level-6/mCkmbxdMsMTjBc3Bm

public class ArrayReplace { 
  int[] arrayReplace(int[] inputArray, int elemToReplace, int substitutionElem) {
    for(int i = 0; i < inputArray.length; i++) {  
        if (inputArray[i] == elemToReplace) 
            inputArray[i] = substitutionElem; 
        else  
            continue;
    } 
    return inputArray; 
  }
}
